[Android] EN有効化時に`ApiException`が出ることがある

## 不具合の内容 / Describe the bug
Google Playのリリース前レポートで報告されたクラッシュ。

ホーム画面でENを有効化するときにApiExceptionが出て落ちている。
Activityで落ちているのか、HomePageViewModel内で落ちているのかは不明（Xamarinなのでこれ以上追えない）。

```
FATAL EXCEPTION: main
Process: jp.go.mhlw.covid19radar, PID: 16774
java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1374)
Caused by: java.lang.reflect.InvocationTargetException
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:327)
	... 1 more
Caused by: com.google.android.gms.common.api.ApiException: 20: The connection to Google Play services was lost due to service disconnection. Last reason for disconnect: DeadObjectException thrown while running ApiCallRunner.
	at com.google.android.gms.common.api.internal.zaaa.zah(com.google.android.gms:play-services-base@@17.6.0:12)
	at com.google.android.gms.common.api.internal.zaaa.zae(com.google.android.gms:play-services-base@@17.6.0:7)
	at com.google.android.gms.common.api.internal.zabl.zaC(com.google.android.gms:play-services-base@@17.6.0:3)
	at com.google.android.gms.common.api.internal.zabl.onConnectionSuspended(com.google.android.gms:play-services-base@@17.6.0:2)
	at com.google.android.gms.common.api.internal.zabl.zaG(com.google.android.gms:play-services-base@@17.6.0:6)
	at com.google.android.gms.common.api.internal.zabl.zaF(com.google.android.gms:play-services-base@@17.6.0:5)
	at com.google.android.gms.common.api.internal.zabl.zaE(com.google.android.gms:play-services-base@@17.6.0:3)
	at com.google.android.gms.common.api.internal.zabl.zaB(com.google.android.gms:play-services-base@@17.6.0:13)
	at com.google.android.gms.common.api.internal.zabl.onConnected(com.google.android.gms:play-services-base@@17.6.0:2)
	at com.google.android.gms.common.internal.zah.onConnected(com.google.android.gms:play-services-base@@17.6.0:1)
	at com.google.android.gms.common.internal.zzf.zza(com.google.android.gms:play-services-basement@@17.6.0:10)
	at com.google.android.gms.common.internal.zza.zzd(Unknown Source:6)
	at com.google.android.gms.common.internal.zzc.zze(com.google.android.gms:play-services-basement@@17.6.0:3)
	at com.google.android.gms.common.internal.zzb.handleMessage(com.google.android.gms:play-services-basement@@17.6.0:31)
	at android.os.Handler.dispatchMessage(Handler.java:105)
	at android.os.Looper.loop(Looper.java:164)
	at android.os.HandlerThread.run(HandlerThread.java:65)
```

## 再現手順 / Steps to reproduce

初期セットアップ時にチュートリアルでENを有効にせず、ホーム画面でENを有効化する。

## 期待される挙動 / Expected behavior

Exceptionが発生しない

## スクリーンショット / Screenshots

<!--
  スクリーンショットがありますと、不具合の説明がより分かりやすくなります。
  If applicable, add screenshots to help explain your problem.
-->

## 動作環境 / Environments

- デバイス：Samsung Galaxy S7 Edge
- OS：Android 8.0
- バージョン：COCOA v2.0.0-beta02

## その他 / Additional context

<!--
  そのほかに補足情報がありましたらこちらに記述してください。
  Add any other context about the problem here.
-->
----
Internal IDs:
- Bug 6928

--- a/Covid19Radar/Covid19Radar.Android/Services/ExposureNotificationApiService.cs
+++ b/Covid19Radar/Covid19Radar.Android/Services/ExposureNotificationApiService.cs
@@ -49,7 +49,10 @@ namespace Covid19Radar.Droid.Services
                 }
                 else
                 {
-                    throw apiException;
+                    throw new AndroidGooglePlayServicesApiException(
+                        apiException.StatusCode,
+                        apiException.Message
+                        );
                 }
             }
         }
@@ -58,7 +61,18 @@ namespace Covid19Radar.Droid.Services
 
         public override async Task<IList<ExposureNotificationStatus>> GetStatusesAsync()
         {
-            return await Client.GetStatusesAsync();
+            try
+            {
+                return await Client.GetStatusesAsync();
+            }
+            catch(ApiException apiException)
+            {
+                throw new AndroidGooglePlayServicesApiException(
+                    apiException.StatusCode,
+                    apiException.Message
+                    );
+            }
+
         }
 
         public override async Task<List<TemporaryExposureKey>> GetTemporaryExposureKeyHistoryAsync()
@@ -77,7 +91,10 @@ namespace Covid19Radar.Droid.Services
                 }
                 else
                 {
-                    throw apiException;
+                    throw new AndroidGooglePlayServicesApiException(
+                        apiException.StatusCode,
+                        apiException.Message
+                        );
                 }
             }
         }
@@ -121,7 +138,10 @@ namespace Covid19Radar.Droid.Services
                 }
                 else
                 {
-                    throw apiException;
+                    throw new AndroidGooglePlayServicesApiException(
+                        apiException.StatusCode,
+                        apiException.Message
+                        );
                 }
             }
         }
@@ -142,7 +162,10 @@ namespace Covid19Radar.Droid.Services
                 }
                 else
                 {
-                    throw apiException;
+                    throw new AndroidGooglePlayServicesApiException(
+                        apiException.StatusCode,
+                        apiException.Message
+                        );
                 }
             }
         }

--- a/Covid19Radar/Covid19Radar/Services/AbsExposureNotificationApiService.cs
+++ b/Covid19Radar/Covid19Radar/Services/AbsExposureNotificationApiService.cs
@@ -11,6 +11,19 @@ using Covid19Radar.Services.Logs;
 
 namespace Covid19Radar.Services
 {
+    public class AndroidGooglePlayServicesApiException : Exception
+    {
+        public readonly int StatusCode;
+
+        public AndroidGooglePlayServicesApiException(
+            int statusCode,
+            string message
+            ) : base(message)
+        {
+            StatusCode = statusCode;
+        }
+    }
+
     public abstract class AbsExposureNotificationApiService : AbsExposureNotificationClient
     {
         private readonly ILoggerService _loggerService;

--- a/Covid19Radar/Covid19Radar/ViewModels/HomePage/HomePageViewModel.cs
+++ b/Covid19Radar/Covid19Radar/ViewModels/HomePage/HomePageViewModel.cs
@@ -202,6 +202,11 @@ namespace Covid19Radar.ViewModels
                 loggerService.Exception("Failed to exposure notification start.", exception);
                 await UpdateView();
             }
+            catch (AndroidGooglePlayServicesApiException exception)
+            {
+                loggerService.Exception("Failed to exposure notification start.", exception);
+                await UpdateView();
+            }
             finally
             {
                 loggerService.EndMethod();
@@ -402,19 +407,27 @@ namespace Covid19Radar.ViewModels
             loggerService.StartMethod();
 
             var daysOfUse = _userDataRepository.GetDaysOfUse();
-
             PastDate = daysOfUse.ToString();
 
-            var statusCodes = await exposureNotificationApiService.GetStatusCodesAsync();
+            bool isStopped;
+            try
+            {
+                var statusCodes = await exposureNotificationApiService.GetStatusCodesAsync();
 
-            var isStopped =
-                statusCodes.Contains(ExposureNotificationStatus.Code_Android.INACTIVATED)
-                || statusCodes.Contains(ExposureNotificationStatus.Code_Android.FOCUS_LOST)
-                || statusCodes.Contains(ExposureNotificationStatus.Code_iOS.Disabled)
-                || statusCodes.Contains(ExposureNotificationStatus.Code_iOS.Unauthorized)
-                || statusCodes.Contains(ExposureNotificationStatus.Code_Android.BLUETOOTH_DISABLED)
-                || statusCodes.Contains(ExposureNotificationStatus.Code_iOS.BluetoothOff)
-                || statusCodes.Contains(ExposureNotificationStatus.Code_Android.LOCATION_DISABLED);
+                isStopped = statusCodes.Contains(ExposureNotificationStatus.Code_Android.INACTIVATED)
+                    || statusCodes.Contains(ExposureNotificationStatus.Code_Android.FOCUS_LOST)
+                    || statusCodes.Contains(ExposureNotificationStatus.Code_iOS.Disabled)
+                    || statusCodes.Contains(ExposureNotificationStatus.Code_iOS.Unauthorized)
+                    || statusCodes.Contains(ExposureNotificationStatus.Code_Android.BLUETOOTH_DISABLED)
+                    || statusCodes.Contains(ExposureNotificationStatus.Code_iOS.BluetoothOff)
+                    || statusCodes.Contains(ExposureNotificationStatus.Code_Android.LOCATION_DISABLED);
+            }
+            catch (AndroidGooglePlayServicesApiException exception)
+            {
+                loggerService.Exception("Failed to exposure notification GetStatusCodesAsync.", exception);
+                isStopped = true;
+            }
+
             var canConfirmExposure = _userDataRepository.IsCanConfirmExposure();
 
             if (isStopped)

